script/format_nix.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+# Executes the Uncrustify code beautifier.
+# Beautification is needed to pass the CI/CD checks for a pull request.
+# This script can be run from any directory within the libspdm repository.
+
+# Check if uncrustify is present.
+if ! command -v uncrustify &> /dev/null
+then
+    echo "ERROR: Unable to execute uncrustify."
+    exit 1
+fi
+
+# Change directory to top of repository.
+cd `dirname $0`
+cd ../
+
+# Exclude non-libspdm submodules.
+EXCLUDE_PATH1="./unit_test/test_size/intrinsiclib/ia32/*"
+EXCLUDE_PATH2="./os_stub/mbedtlslib/mbedtls/*"
```

Comment:
I think we can use `git ls-files | sed "/unit_test\/test_size\/intrinsiclib\/ia32/d" | sed -n '/.\.[ch]$/p' | uncrustify -q -c ./.uncrustify.cfg --replace --no-backup -F -`. This way we don't have to explicitly mention the submodules and is forward compatible with more submodules being added.

Reply:
That's modifying files in `mbedtls/` and `openssl/`. How about I keep this as-is and then we can refactor it later.

Reply:
Really? That is surprising! There is a specific `--recurse-submodules` option to recurse submodules and I didn't see it modify them on my testing :(.